src/components/Checkbox/index.js

```diff
@@ -0,0 +1,53 @@
+import { Component } from 'pet-dex-utilities';
+import './index.scss';
+
+const events = ['change'];
+
+const html = `
+    <label class="checkbox-container">
+        <input data-select="checkbox" type="checkbox" value="">
+        <span data-select="checkbox-text" class="checkmark"></span>
+    </label>
+`;
+
+export default function Checkbox({ isChecked = false, text = '', isDisabled = false, value = '' }) {
+    Component.call(this, { html, events });
+
+    this.setChecked(isChecked);
+    this.setText(text);
+    this.setDisabled(isDisabled);
+    this.setValue(value);
+
+    const $checkbox = this.selected.get('checkbox');
+    const $checkboxText = this.selected.get('checkbox-text');
+
+    $checkbox.addEventListener('change', () => {
+        this.emit('change', $checkbox.checked);
+    });
```

Comment:
Na maior parte dos casos a gente sempre vai preparar os dados no listener e enviar para um método do componente. Pois quem lida com a lógica é o método

```suggestion
    $checkbox.addEventListener('change', (e) => {
        const isChecked = e.target.checked;
        this.setChecked(isChecked);
    });
```